sql/connect/server/src/main/scala/org/apache/spark/sql/connect/ml/MLUtils.scala

```diff
@@ -269,8 +293,9 @@ private[ml] object MLUtils {
    * @return
    *   the ML instance
    */
-  def load(className: String, path: String): Object = {
-    val loadedMethod = SparkClassUtils.classForName(className).getMethod("load", classOf[String])
+  def load(sessionHolder: SessionHolder, className: String, path: String): Object = {
+    val name = replaceOperator(sessionHolder, className)
+    val loadedMethod = SparkClassUtils.classForName(name).getMethod("load", classOf[String])
```

Comment:
this is the only place where we don't use the service loader is there a way to change this? Maybe by trying first estimator and then transformer? Loading any class here is not an ideal situation.

Reply:
Hi @grundprinzip 

> this is the only place where we don't use the service loader is there a way to change this?

For estimators/evaluators, we can use ServiceLoader to load the corresponding class here and call the static "load" function. But for models, ServiceLoader is not going to work given that some Models don't have **a zero-argument constructor** which is required by ServiceLoader. We could add ** zero-argument constructor ** for the model to make it work?

> Maybe by trying first estimator and then transformer? 
I couldn't follow up

Reply:
Wow, just tried adding default zero-argument constructor, but ServiceLoader requires it to be public. or else, ServiceLoader will throw exception.

Reply:
Probably something for a follow up, but is it hard to add the public no-arg constructor?

Reply:
Hi @grundprinzip, Let's take LogisticRegressionModel as an example, you can see from https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala#L1062, 

If we would like to add a public no-arg constructor, then we need to set the default other parameters. but the question is what values of the parameters are ok?

Let me file an issue to track it.

Reply:
I filed an issue https://issues.apache.org/jira/browse/SPARK-50844 to track it.

Reply:
Can you add an assertion that the following trait is implemented: https://github.com/apache/spark/blob/97241eb0cfc98f8911dc205d7a19f31a0eaad5fb/mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala#L342

The class must support MLReadable at least as the basic precaution.

Reply:
Thanks!